src/common/eeprom.cpp

```diff
@@ -623,120 +627,127 @@ static int eeprom_convert_from_v2(void) {
+/**
+ * @brief conversion function for old version 6 (v 4.1.0)
+ *
+ * @param eevars input/output eeprom struct
+ */
+static void eeprom_convert_from_v6(eeprom_vars_t &eevars) {
     eeprom_vars_t vars = eeprom_var_defaults;
     eeprom_init_FW_identifiers(vars);
 
-    // start addres of imported data block (FILAMENT_TYPE..EEVAR_SOUND_MODE)
-    eeprom_import_block(EEVAR_FILAMENT_TYPE, EEVAR_SOUND_VOLUME, &(vars.FILAMENT_TYPE));
-
-    eeprom_make_patches(vars);
-
-    eeprom_save_upgraded(vars);
-    return 1;
+    eeprom_import_block(EEVAR_FILAMENT_TYPE, EEVAR_SOUND_VOLUME, vars, eevars);
+    eevars = vars;
 }
 
-// conversion function for old version 8 (v 4.2.x)
-static int eeprom_convert_from_v7(void) {
+/**
+ * @brief conversion function for old version 8 (v 4.2.x)
+ *
+ * @param eevars input/output eeprom struct
+ */
+static void eeprom_convert_from_v7(eeprom_vars_t &eevars) {
     eeprom_vars_t vars = eeprom_var_defaults;
     eeprom_init_FW_identifiers(vars);
 
-    // start addres of imported data block (FILAMENT_TYPE..EEVAR_LANGUAGE)
-    eeprom_import_block(EEVAR_FILAMENT_TYPE, EEVAR_FILE_SORT, &(vars.FILAMENT_TYPE));
-
-    eeprom_make_patches(vars);
-
-    eeprom_save_upgraded(vars);
-    return 1;
+    eeprom_import_block(EEVAR_FILAMENT_TYPE, EEVAR_FILE_SORT, vars, eevars);
+    eevars = vars;
 }
 
-// conversion function for old version 8 (v 4.3.RC)
-static int eeprom_convert_from_v8(void) {
+/**
+ * @brief conversion function for old version 8 (v 4.3.RC)
+ *
+ * @param eevars input/output eeprom struct
+ */
+static void eeprom_convert_from_v8(eeprom_vars_t &eevars) {
     eeprom_vars_t vars = eeprom_var_defaults;
     eeprom_init_FW_identifiers(vars);
 
-    // start addres of imported data block (FILAMENT_TYPE..EEVAR_SOUND_MODE)
-    eeprom_import_block(EEVAR_FILAMENT_TYPE, EEVAR_MENU_TIMEOUT, &(vars.FILAMENT_TYPE));
-
-    eeprom_make_patches(vars);
-
-    eeprom_save_upgraded(vars);
-    return 1;
+    eeprom_import_block(EEVAR_FILAMENT_TYPE, EEVAR_MENU_TIMEOUT, vars, eevars);
+    eevars = vars;
 }
 
-// conversion function for old version 9 (v 4.3.2)
-static int eeprom_convert_from_v9(void) {
+/**
+ * @brief conversion function for old version 9 (v 4.3.2)
+ *
+ * @param eevars input/output eeprom struct
+ */
+static void eeprom_convert_from_v9(eeprom_vars_t &eevars) {
     eeprom_vars_t vars = eeprom_var_defaults;
     eeprom_init_FW_identifiers(vars);
 
-    eeprom_import_block(EEVAR_FILAMENT_TYPE, EEVAR_DEVHASH_IN_QR, &(vars.FILAMENT_TYPE));
-
-    eeprom_make_patches(vars);
-
-    eeprom_save_upgraded(vars);
-    return 1;
-}
+    eeprom_import_block(EEVAR_FILAMENT_TYPE, EEVAR_DEVHASH_IN_QR, vars, eevars);
+    eevars = vars;
+}
+
+/**
+ * @brief conversion function for new version format (features, firmware version/build)
+ * does not change crc, it is changed automatically by write function
+ *
+ * @param eevars eeprom struct in RAM
+ * @return true updated (changed)
+ * @return false not changed
```

Comment:
suggest @retval instead of @return

Reply:
it is generated automatically this way

Reply:
via _Doxygen Documentation Generator_
vscode plugin